### PR TITLE
Add selector-max-pseudo-class rule

### PR DESCRIPTION
**Pull request by Bilie**
> Which issue, if any, is this issue related to?

#2805 

> Is there anything in the PR that needs further explanation?

I had to use --no-verify flag when committing, as I was getting:
✖ npm run prettier --write found some errors. Please fix them and try committing again.
But no errors were listed and I did not know how to fix this.

### CODE REVIEW

File: lib/rules/selector-max-pseudo-class/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## Options
+
+`int`: Maximum pseudo classes allowed.
+
+For example, with `1`:
+
+The following patterns are considered violations:
+
+```css
+a:first:focus { color: pink; }
```
Comment (jeddy3) [inline lib/rules/selector-max-pseudo-class/README.md]: Can we remove the declarations, please? And just have an empty block `{}`.

File: lib/rules/selector-max-pseudo-class/__tests__/index.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  reject: [
+    {
+      code: ":global {}",
+      message: messages.expected(":global", 0)
+    },
+    {
+      code: "a:first-child { color: pink; }",
+      message: messages.expected("a:first-child", 0)
+    }
+  ]
```
Comment (jeddy3) [inline lib/rules/selector-max-pseudo-class/__tests__/index.js]: Can we please get a `reject` test for a selector on the 2nd line of a selector list? e.g.

```
a,\na:first-child {}
```

File: lib/rules/selector-max-pseudo-class/__tests__/index.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+      code: "a { color: pink; }",
+      description: "selector with no pseudo class"
+    },
+    {
+      code: ".foo { .bar { .baz {} } }",
+      description: "nested selectors with no pseudo classes"
+    }
+  ],
+
+  reject: [
```
Comment (jeddy3) [inline lib/rules/selector-max-pseudo-class/__tests__/index.js]: Can we add `line` and `column` properties and values to all the `reject` tests, please? e.g.

```
{
  code: ".foo { .bar { .baz {} } }",
  description: "nested selectors with no pseudo classes"
  line: 1,
  column: 1
}
```

**Comment by jeddy3** [comment]
@Bilie Friendly ping to see if you're still working on this PR?

**Comment by Bilie** [comment]
Hi @jeddy3 Thank you very much for all your feedback, it was really helpful, and for the ping :) I have updated the PR, I hope all issues you mentioned have been resolved!

File: lib/rules/selector-max-pseudo-class/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+`int`: Maximum pseudo classes allowed.
+
+For example, with `1`:
+
+The following patterns are considered violations:
+
+```css
+a:first:focus {}
+
```
Comment (jeddy3) [inline lib/rules/selector-max-pseudo-class/README.md]: Please remove empty line

File: lib/rules/selector-max-pseudo-class/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+```css
+.foo .bar:first-child:hover {}
+
```
Comment (jeddy3) [inline lib/rules/selector-max-pseudo-class/README.md]: Please remove empty line

File: lib/rules/selector-max-pseudo-class/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+Limit the number of pseudo classes in a selector.
+
+```css
+.foo .bar:first-child:hover {}
+/*       ↑           ↑
+         |           |
+         1           2 -- this selector contains two pseudo classes */
+```
+
```
Comment (jeddy3) [inline lib/rules/selector-max-pseudo-class/README.md]: We should add same description from the other `selector-max-*` rules here:

```
This rule resolves nested selectors before counting the number of pseudo-classes in a selector. Each selector in a [selector list](https://www.w3.org/TR/selectors4/#selector-list) is evaluated separately.

The content of the `:not()` pseudo-class is also evaluated separately. The rule processes the argument as if it were an independent selector, and the result does not count toward the total for the entire selector.
```

File: lib/rules/selector-max-pseudo-class/README.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+## Options
+
+`int`: Maximum pseudo classes allowed.
+
+For example, with `1`:
+
+The following patterns are considered violations:
+
+```css
+a:first:focus {}
```
Comment (jeddy3) [inline lib/rules/selector-max-pseudo-class/README.md]: `first-child`

File: lib/rules/selector-max-pseudo-class/__tests__/index.js [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+      code: "a {}",
+      description: "empty rule"
+    },
+    {
+      code: '@import "foo.css";',
+      description: "blockless statement"
+    },
+    {
+      code: "a {}",
+      description: "selector with no pseudo class"
```
Comment (jeddy3) [inline lib/rules/selector-max-pseudo-class/__tests__/index.js]: Please hyphenate [pseudo-classes](https://www.w3.org/TR/selectors-3/#pseudo-classes) in all the descriptions.

File: lib/rules/selector-max-pseudo-class/__tests__/index.js [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+    },
+    {
+      code: ".foo { .bar { .baz {} } }",
+      description: "nested selectors with no pseudo classes"
+    }
```
Comment (jeddy3) [inline lib/rules/selector-max-pseudo-class/__tests__/index.js]: Can we get two simple pseudo-element tests here please, for lines 43-50 of the rule code? e.g.

``` 
{
  code: "a:before {}",
  description: "single level 2 puesdo-element"
},
{
  code: "a::before {}",
  description: "single level 3 puesdo-element"
}
```

**Comment by jeddy3** [comment]
> -   Added: `selector-max-pseudo-class` rule ([#3195](https://github.com/stylelint/stylelint/pull/3195)).

**Pull request merged**